packages/manager/apps/dedicated/client/app/ip/components/list/firewall/game/ip-ip-firewall-game.controller.js

```diff
@@ -294,6 +294,86 @@ export default /* @ngInject */ function IpGameFirewallCtrl(
     });
   }
 
+  // Check if the new rule is not into other rules
+  // for example:
+  // newRule = {from: 4320, to: 4330},
+  // rule = {from: 4300, to: 4400}
+  function hasNewRuleIntoRule(newRule, rule) {
+    return (
+      newRule.ports.from < rule.ports.from && newRule.ports.to < rule.ports.to
+    );
+  }
+
+  // Check if the new rule port to is not into other rules
+  // for example:
+  // newRule = {from: 4200, to: 4330},
+  // rule = {from: 4300, to: 4400}
+  function hasNewRulePortToIntoRule(newRule, rule) {
+    return (
+      newRule.ports.from < rule.ports.to &&
+      newRule.ports.to > rule.ports.from &&
+      newRule.ports.to < rule.ports.to
+    );
+  }
+
+  // Check if the new rule port from is not into other rules
+  // for example:
+  // newRule = {from: 4300, to: 4500},
+  // rule = {from: 4200, to: 4400}
+  function hasNewRulePortFromIntoRule(newRule, rule) {
+    return (
+      newRule.ports.from > rule.ports.from &&
+      newRule.ports.from < rule.ports.to &&
+      newRule.ports.to > rule.ports.to
+    );
+  }
+
+  // Check if the rule is not into new rule
+  // for example:
+  // newRule = {from: 4100, to: 4500},
+  // rule = {from: 4200, to: 4300}
+  function hasRuleIntoNewRule(newRule, rule) {
+    return (
+      newRule.ports.from < rule.ports.from && newRule.ports.to > rule.ports.to
+    );
+  }
+
+  function hasPortsAlreadyUsed() {
+    return self.table.rules.some((rule) => {
+      if (rule.ports.from === rule.ports.to) {
+        if (self.rule.ports.from === self.rule.ports.to) {
+          if (self.rule.ports.from === rule.ports.from) {
+            return true;
+          }
+        } else if (
+          self.rule.ports.from < self.rule.ports.to &&
+          rule.ports.from > self.rule.ports.from
+        ) {
+          return true;
+        }
+      } else if (self.rule.ports.from === self.rule.ports.to) {
+        if (
+          self.rule.ports.from < rule.ports.to &&
+          self.rule.ports.from > rule.ports.from
+        ) {
+          return true;
```

Comment:
you can also split it in sub functions or at least comment what are the different cases